[8.x] Fix appendable attributes in Blade components

panda-madness opened the pull request:
Currently, attempting to render a component which prepends a non-class attribute in it's template without actually overriding it in the calling template yields an Exception.

For example, given the following component called `test-component`:

```blade
<div {{ $attributes->merge(['data-controller' => $attributes->prepends('profile-controller')]) }}>
    {{ $slot }}
</div>
```

Rendering this component like so: `<x-test-component data-controller="some-controller" />` works as expected.

Rendering this component like so: `<x-test-component />` yields the following error:

<img width="1209" alt="Screen Shot 2020-11-06 at 13 59 09" src="https://user-images.githubusercontent.com/6180087/98340888-41417b80-2038-11eb-852c-a6815f5f7b72.png">

This PR fixes this behavior. `<x-test-component />` will now properly render as
```
<div data-controller="profile-controller"></div>
```

See #35128 for more details.

panda-madness committed:
Update AppendableAttributeValue.php

panda-madness committed:
Add tests

driesvints commented on src/Illuminate/View/AppendableAttributeValue.php:
```diff
@@ -21,4 +21,9 @@ public function __construct($value)
+
+    public function __toString()
+    {
+        return (string)$this->value;
```
```suggestion
        return (string) $this->value;
```

driesvints commented on src/Illuminate/View/AppendableAttributeValue.php:
```diff
@@ -21,4 +21,9 @@ public function __construct($value)
         $this->value = $value;
     }
+
+    public function __toString()
```
Add a docblock in the same formatting as the rest of the framework.

panda-madness committed:
Fix tests

panda-madness committed:
Add docblock

driesvints commented:
@panda-madness it's also best that you provide a thorough explanation in your PR description (see our PR templates) so Taylor knows what this is about and why it's needed. Don't just link to an issue.

panda-madness committed:
Fix unrelated test

panda-madness commented:
@driesvints I've updated the PR description.

I've also modified the test to include this use case, but I'm not quite sure why an unrelated test started failing...

driesvints committed:
Update hello-span.blade.php

driesvints committed:
Update BladeTest.php

driesvints commented:
@panda-madness your whitespace change from here seems to be the culprit: https://github.com/laravel/framework/pull/35131/commits/8b2e529a1e4c6fc8dc4138f6d22b1a5798e4b588#diff-9e28a6d9f9f352efc3960eab4ff2d081a25459d765d3f1ac2c9735dd4356dd50L45

Think I got it now.

panda-madness commented:
This is a little nitpicky, but should we change the `AppendableAttributeValue` class name to `PrependableAttributeValue` so it maches the method name better?

taylorotwell committed:
Update AppendableAttributeValue.php

driesvints commented:
@panda-madness probably not worth the breaking change.
